mart/models/modular.py

```diff
@@ -80,6 +78,13 @@ def __init__(
         self.test_step_log = test_step_log or []
         self.test_metrics = test_metrics
 
+        # Load state dict for specified modules
+        load_state_dict = load_state_dict or {}
+        for name, path in load_state_dict.items():
+            module = attrgetter(name)(self.model)
```

Comment:
Though it gets nested attributes now, I'm not sure how to add a name with dots in the Hydra command  line.

Reply:
Ah `attrgetter` is a nice find! Looks like we'd have to change the separator for it to work on the commandline.

Reply:
I added a `get_dottedpath_items()` to extract dotted paths from nested dictionaries. So we can keep the command line tiny.

Reply:
Can you give me an example command line? Trying to understand the code you wrote.

Reply:
Okay now I get what you implemented: flatten. Can you rename this to flatten. One issue, however, is that your implementation will not fail in the case of two keys with the same name. For example:
```json
{
  "logits": { "0": "0.pt" },
  "logits.0": "logits_0.pt",
}
```

I feel like that should throw an error.

Reply:
I found a package https://flatdict.readthedocs.io/en/stable/#docs
, but it does not detect key collision either.

```python
>>> from flatdict import FlatDict
>>> FlatDict({"a": {"b": "c.pth"}}, delimiter=".")["a.b"]
'c.pth'
>>> FlatDict({"a": {"b": "c.pth"}, "a.b": "d.pth"}, delimiter=".")["a.b"]
'd.pth'
```

Reply:
`pandas` does not detect key collision either.

```python
>>> import pandas as pd

>>> d = {'a': 1,
     'c.a': 1,
     'c': {'a': 2, 'b': {'x': 5, 'y' : 10}},
     'd': [1, 2, 3]}

>>> df = pd.json_normalize(d, sep='.')

>>> print(df.to_dict(orient='records')[0])
{'a': 1, 'c.a': 2, 'd': [1, 2, 3], 'c.b.x': 5, 'c.b.y': 10}
```

Reply:
I implemented `mart.utils.utils.FlattenDict` that detects key collision. Two test cases are added.